src/Features/Core/Portable/Wrapping/ChainedExpression/AbstractChainedExpressionWrapper.cs

```diff
@@ -271,39 +273,56 @@ private bool IsDecomposableChainPart(SyntaxNode node)
         }
 
         /// <summary>
-        /// Recursively walks down <paramref name="node"/> decomposing it into the individual 
-        /// tokens and nodes we want to look for chunks in. 
+        /// Walks down <paramref name="node"/> decomposing it into the individual tokens
+        /// and nodes we want to look for chunks in.
         /// </summary>
         private void Decompose(SyntaxNode node, ArrayBuilder<SyntaxNodeOrToken> pieces)
         {
-            // Ignore null nodes, they are never relevant when building up the sequence of
-            // pieces in this chained expression.
-            if (node is null)
+            // Recursion can get us into trouble with deeply chained statements. Let's
+            // use a queue instead.
+            var queue = SharedPools.Default<Queue<SyntaxNode>>().AllocateAndClear();
```

Comment:
this seems wrong.  the previous logic would dive into the child if it was a node.  indicating that this was a DFS type walk.  the current logic puts it at the end, indicating more of a BFS approach to the nodes.

Reply:
Yes, that makes sense. I was just happy to see the crash gone.

I'll convert it to use a stack and do DFS later when I get some more time.

Reply:
Done. Note that while this solves the crashing problem in AbstractChainedExpressionWrapper.Decompose(), it still reports an out of stack problem elsewhere. At least this stops it from unceremoniously crashing and dumping you out, though.

Note that this still builds the items in the reverse order as before. I added a reverse at the end for that. In my extreme scenario of 32K lines, we wind up with an ArrayBuilder collection of 98K objects to reverse. That happens pretty fast, and this isn't called often, so I think that's a good trade to get rid of the crash.